_posts/2024-12-30-debezium-oracledb-logminer.md

```diff
@@ -0,0 +1,313 @@
+
+  '''
+  $> docker compose exec oracle bash
+  $> sqlplus ‘/ as sysdba’
+
+  SQL> SELECT LOG_MODE FROM V$DATABASE
+
+  '''
+
+  If the column contains ARCHIVELOG, then archive logging is enabled. If the column contains the value NOARCHIVELOG, archive logging isn’t enabled, and further configuration is necessary.
+
+  Execute the following SQL commands inside the SQL*Plus terminal window:
+  
+  '''
+  ALTER SYSTEM SET db_recovery_file_dest_size = 10G;
+  ALTER SYSTEM SET db_recovery_file_dest = '/opt/oracle/oradata/ORCLCDB' scope=spfile;
+  SHUTDOWN IMMEDIATE
+  STARTUP MOUNT
+  ALTER DATABASE ARCHIVELOG;
+  ALTER DATABASE OPEN;
+  ARCHIVE LOG LIST;
+  '''
+  Output of last executed SQL command should show Archive mode for Database log mode.
+  '''
+  Database log mode                  Archive Mode
+  Automatic archival                 Enabled
+  Archive destination                USE_DB_RECOVERY_FILE_DEST
+  Oldest online log sequence         1
+  Next log sequence to archive       3
+  Current log sequence               3
+  '''
+
+
+
+# Redo Logs
+
+  Oracle Redo logs are the transactional logs. Using the same terminal window, execute below listed SQL command:
+
+  '''
+  SQL> SELECT GROUP#, MEMBER FROM V$LOGFILE ORDER BY 1, 2;
+
+  GROUP# MEMBER
+  ---------- ---------------------------------------------------
+    1 /opt/oracle/oradata/ORCLCDB/redo01.log
+    2 /opt/oracle/oradata/ORCLCDB/redo02.log
+    3 /opt/oracle/oradata/ORCLCDB/redo03.log
+  
+  
+  ALTER DATABASE CLEAR LOGFILE GROUP 1; 
+  ALTER DATABASE DROP LOGFILE GROUP 1; 
+  ALTER DATABASE ADD LOGFILE GROUP 1 ('/opt/oracle/oradata/ORCLCDB/redo01.log') size 400M REUSE;
+  ALTER SYSTEM SWITCH LOGFILE;
+
+  '''
+
+# Supplemental Logging
+
+  Database supplementary logging needs to be enabled at the very least for Debezium to communicate with LogMiner, handle chained rows, and work with different storage configurations.
+
+  '''
+  SQL> ALTER DATABASE ADD SUPPLEMENTAL LOG DATA
+  '''
+
+
+# Users Setup in Oracle DB
+
+  To capture change events, the Debezium connector needs to establish a JDBC connection to the Oracle database and interact with LogMiner APIs. This requires a dedicated database user account with specific permissions to access LogMiner and read data from the target tables.
+
+  In the same SQLplus terminal window, create table spaces:
+
+
+  '''
+  CONNECT sys/oraclepw@ORCLCDB as sysdba; 
+  CREATE TABLESPACE logminer_tbs DATAFILE '/opt/oracle/oradata/ORCLCDB/logminer_tbs.dbf' SIZE 25M REUSE AUTOEXTEND ON MAXSIZE UNLIMITED; 
+
+  CONNECT sys/oraclepw@ORCLPDB1 as sysdba; 
+  CREATE TABLESPACE logminer_tbs DATAFILE '/opt/oracle/oradata/ORCLCDB/ORCLPDB1/logminer_tbs.dbf' SIZE 25M REUSE  AUTOEXTEND ON MAXSIZE UNLIMITED;
+  '''
+
+> NOTE – Replace oraclepw with the password set in docker compose file.
```

Comment:
Looks like there are other commands being run apart from replacing the password.